### DISCUSSION TITLE
Entry Points : MSIPackagingTool.exe

### DISCUSSION THREAD

**Discussion by JPH71** [Microsoft/msix-packaging#618] [General]
I have an MSIX that I have created by commandline
How do I ensure it uses the correct executable when it first launches?
I have tried the following but it is no longer a valid parameter...
`MsixPackagingTool.exe package -p "Path_toMSIX" --entry-point "Wanted.exe"`
`MsixPackagingTool.exe package -p "Path_toMSIX" --remove-entry-point "NotWanted.exe"`
`MsixPackagingTool.exe entry-point -p "Path_toMSIX" -e "Wanted.exe"`
`MsixPackagingTool.exe remove-entry-point -p "Path_toMSIX" -e "NotWanted.exe"`

The MsixPackagingTool.exe only uses the following as parameters when you ask for help:
 cleanup         Cleans up the tool's temporary files and artifacts.
 create-package  Creates a MSIX package from an application's command-line installer.
 get-status      Gets the status of a conversion machine.

But Copilot gave the commands I used...

I guess the tool has changed and these are no longer offered....

Surely there must be a way?

**Comment by florelis** [comment]
I'm not an expert in Copilot and it has been a long time since I last used MSIX Packaging Tool, but I think Copilot may have made up these parameters.

I don't know if there is something for removing entry points, but you can use a [template](https://learn.microsoft.com/windows/msix/packaging-tool/generate-template-file) for the conversion and specify entrypoints to add in it.

**Comment by JPH71** [comment]
I was coming to that conclusion - I tried to find an example for setting up
entry points in the MSIX packaging tool template - but these did not seem
to work.
If you ever come across an example, please forward it to me.

Many thanks for taking the time to respond.

**Comment by Vikrant-chib** [comment]
When MSIX app is installed and then launched, it will use the first entry point listed in the AppxManifest.
 
You can view the entry point by opening the MSIX Package with the MSIX Packaging Tool. Use the Package Editor to open the manifest file and see the entry point defined for your package. This information is found in the package's manifest file (`AppxManifest.xml`) under the `<Applications>` element. Within this section, each `<Application>` element may have an `<EntryPoint>` sub-element that specifies the executable to be run.
 
Moreover, during the conversion process, you can select detected entry points from the 'First Launch Package' page of the MSIX Packaging Tool UI. If the package is already created, you can directly edit the AppxManifest using the Package Editor.

**Comment by JPH71** [comment]
I am using the msixpackaging tool in command line mode...

I know that with the Gui you can set the entry point by running first
launch tasks...

I am packaging MECM/sccm Configuration manager... but it picks the report
viewer as the default...

I tried editing the appxmanifest file and tried to switch the entries...
but this does not work as when the msix is installed, it does not launch
either of the two executables....

When it's done using the GUI, it works fine as you can select the default
program...

It would be great if this could be done using the msix template file... as
there seem to be entries there... but theses also do not work... and there
are no examples of using the application nodes in that xml file...

**Comment by Vikrant-chib** [comment]
Manually editing manifest generated from CLI should have worked.
Perhaps you could try debugging by comparing the APPX manifests created from the MSIX Packaging Tool UI with one from the CLI.

Application tag mentioned in the [documentation](https://learn.microsoft.com/windows/msix/packaging-tool/generate-template-file) will add application to AppxManifest only if application with same executable name is detected.
